plugins/flat-object/src/test/java/org/opensearch/flatobject/mapper/FlatObjectFieldMapperTests.java

```diff
@@ -6,63 +6,79 @@
  * compatible open source license.
  */
 
-package org.opensearch.flatobject.mapper;
+package org.opensearch.flatobject.query;
 
+import org.apache.lucene.search.Query;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
 import org.opensearch.common.xcontent.XContentBuilder;
-import org.opensearch.flatobject.FlatObjectPlugin;
-import org.opensearch.index.mapper.FieldMapperTestCase2;
-import org.opensearch.plugins.Plugin;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.index.query.AbstractQueryBuilder;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.index.query.QueryShardContext;
 
 import java.io.IOException;
-import java.util.Collection;
 
-public class FlatObjectFieldMapperTests extends FieldMapperTestCase2<FlatObjectFieldMapper.Builder> {
-    private static final String FIELD_TYPE = "flat-object";
+public class FlatObjectQueryBuilder extends AbstractQueryBuilder<FlatObjectQueryBuilder> {
```

Comment:
@mingshl This is breaking java compilation.

The java class name and the file name need to be in sync.
```
OpenSearch-Mingshl/plugins/flat-object/src/test/java/org/opensearch/flatobject/mapper/FlatObjectFieldMapperTests.java:22: error: class FlatObjectQueryBuilder is public, should be declared in a file named FlatObjectQueryBuilder.jav
```

Reply:
Also the `package` path (line no.9) seems the be incorrect.

Reply:
Updated in @mingshl
[Update FlatObjectFieldMapperTests](https://github.com/mingshl/OpenSearch-Mingshl/pull/3/commits/ba34d41761324789951ec0deaa556de0a3fd56f6). all tests should pass now